functions/src/requests/index.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  Allow,
+  IsEnum,
+  IsInt,
+  IsNotEmpty,
+  IsObject,
+  IsString,
+  Max,
+  Min,
+  validate,
+  ValidateNested,
+} from 'class-validator';
+import * as admin from 'firebase-admin';
+import { firestore } from 'firebase-admin';
+import * as functions from 'firebase-functions';
+import { Change, EventContext } from 'firebase-functions/lib/cloud-functions';
+import { DocumentSnapshot } from 'firebase-functions/lib/providers/firestore';
+
+import { IUser, User } from '../users';
+import GeoPoint = admin.firestore.DocumentReference;
+
+export enum RequestStatus {
+  pending = 'pending',
+  ongoing = 'ongoing',
+  completed = 'completed',
+  cancelled = 'cancelled',
+  removed = 'removed'
+}
+
+export interface IRequest extends FirebaseFirestore.DocumentData {
+  cavUserRef: FirebaseFirestore.DocumentReference<IUser> | null;
+  pinUserRef: FirebaseFirestore.DocumentReference<IUser>;
+  pinUserSnapshot: IUser;
```

Comment:
> We want PINs to fill in a questionnaire about themselves (e.g. age, other things that make them vulnerable etc...)

I don't think that's an issue. I see a world where PIN/CAV is a workflow, not necessarily a user distinction. 

I might be a nurse helping elderly in their homes, but I will need someone to walk my dogs.

Reply:
> In the future, not every PIN may have an associated IUser (if a request is being filed on behalf of them by someone else).

*As User I can give someone access to file PIN requests on my behalf.*

Again the goal here is to make sure that we all access the system in a uniform way.

Reply:
> Clearer separation of auth (and other IUser / PII) data from subject data (I have a feeling this will be useful to us later).

So the auth data goes into a separate database (including all PII). The `users` collection should have no PII. Even if we store address we tie it to the user record which includes no PII so it's segregated.